src/test/java/jp/co/yahoo/yosegi/reader/TestYosegiReader.java

```diff
@@ -0,0 +1,307 @@
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    Configuration writerConfig = new Configuration();
+    writerConfig.set( "block.size" , Integer.toString( blockSize ) );
+    YosegiWriter writer = new YosegiWriter( out , writerConfig );
+    List<ColumnBinary> list = Arrays.asList( createStringPushdownTestColumn() );
+    // Block-1 Spread-1
+    writer.appendRow( list , 4 );
+    // Block-1 Spread-2
+    writer.appendRow( list , 4 );
+    writer.writeFixedBlock();
+    // Block-2 Spread-1
+    writer.appendRow( list , 4 );
+    // Block-2 Spread-2
+    writer.appendRow( list , 4 );
+    writer.writeFixedBlock();
+    // Block-3 Spread-1
+    writer.appendRow( Arrays.asList( createStringPushdownTestColumn2() ) , 4 );
+
+    writer.close();
+    out.close();
+    return out.toByteArray();
+  }
+
+  @Test
+  public void T_read_1() throws IOException {
+    byte[] blocks = createTestBinary();
+    ByteArrayInputStream in = new ByteArrayInputStream( blocks );
+    YosegiReader reader = new YosegiReader();
+    reader.setNewStream( in , blocks.length , new Configuration() );
+    assertEquals( reader.hasNext() , true );
+
+    // Block-1 Spread-1
+    List<ColumnBinary> raw = reader.nextRaw();
+    assertEquals( reader.getCurrentSpreadSize().intValue() , 4 );
+    assertEquals( raw.size() , 1 );
+    // Block-1 Spread-2
+    raw = reader.nextRaw();
+    assertEquals( reader.getCurrentSpreadSize().intValue() , 4 );
+    assertEquals( raw.size() , 1 );
+
+    // Block-2 Spread-1
+    raw = reader.nextRaw();
+    assertEquals( reader.getCurrentSpreadSize().intValue() , 4 );
+    assertEquals( raw.size() , 1 );
+    // Block-2 Spread-2
+    raw = reader.nextRaw();
+    assertEquals( reader.getCurrentSpreadSize().intValue() , 4 );
+    assertEquals( raw.size() , 1 );
+
+    // Block-3 Spread-1
+    raw = reader.nextRaw();
+    assertEquals( reader.getCurrentSpreadSize().intValue() , 4 );
+    assertEquals( raw.size() , 1 );
+
+    assertEquals( reader.hasNext() , false );
+  }
+
+  @Test
+  public void T_EmptyPushdown_1() throws IOException {
+    byte[] blocks = createTestBinary();
+    ByteArrayInputStream in = new ByteArrayInputStream( blocks );
+    YosegiReader reader = new YosegiReader();
+    AndExpressionNode index = new AndExpressionNode();
+    index.addChildNode(
+        new ExecuterNode( new StringExtractNode( "column" )
+        , new PerfectMatchStringFilter( "p" ) )
+    );
+    Configuration readerConfig = new Configuration();
+    // Skip Block-1, Block-2
+    reader.setBlockSkipIndex( index );
+    reader.setNewStream( in , blocks.length , readerConfig );
+    assertEquals( reader.hasNext() , true );
+
+    // Block-3 Spread-1
+    List<ColumnBinary> raw = reader.nextRaw();
+    assertEquals( reader.getCurrentSpreadSize().intValue() , 4 );
+    assertEquals( raw.size() , 1 );
+    assertEquals( reader.hasNext() , false );
+  }
+
+  @Test
+  public void T_EmptyPushdown_2() throws IOException {
```

Comment:
please describe what the code wants to test in the method name instead of number extension.